lib/dk_buttons_and_leds/dk_buttons_and_leds.c

```diff
@@ -158,10 +158,10 @@ static void buttons_scan_fn(struct k_work *work)
 	last_button_scan = button_scan;
 
 	if (button_scan != 0) {
-		int err = k_delayed_work_submit(&buttons_scan,
+		int err = k_work_reschedule(&buttons_scan,
 		  K_MSEC(CONFIG_DK_LIBRARY_BUTTON_SCAN_INTERVAL));
 
-		if (err) {
+		if (err < 0) {
```

Comment:
As far as I know, reschedule no longer returns errors, just variations of success, and this check will never be true.

Here and below

@pabigot can you confirm this?

Reply:
fixed